grails-app/services/com/unifina/service/PermissionService.groovy

```diff
@@ -519,7 +570,14 @@ class PermissionService {
 		}
 
 		// Special case of UI channels: they inherit permissions from the associated canvas
-		if (p.empty && resource instanceof Stream && resource.uiChannel) {
+		if (p.isEmpty() && resource instanceof Stream && resource.uiChannel) {
+			//if (hasPermission(userish, resource.uiChannelCanvas, op)) {
```

Comment:
Here we could make this more DRY by leveraging the `getPermissionsTo(resource, userish)` and checking if the returned `List` contains the operation `op`.